### PR TITLE
Add definitions to glossary

### LINKED ISSUE #159: Populate glossary
There are many missing terms. We should fill those in over time!

### PR DESCRIPTION
**Pull request by alice-i-cecile**
Fixes #159, #151.

Currently incomplete: I'd like to see if I can finish this in one fel swoop. Please complain about things that are unclear or seem unintuitive!

### CODE REVIEW

File: design/src/glossary.md [hunk 1]
```diff
@@ -11,34 +11,212 @@ Key concepts and terms from game design.
...
 
+Repetitive work that is done automatically by in-game entities and systems.
+
 ### Challenge
 
+Difficulty that is fun and rewarding for players to overcome.
+
+### Cohesion
+
+The degree to which game elements (usually [thematic](glossary.md#theme) or [art](glossary.md#art-style)) feel like they belong together as part of a greater whole.
```
Comment (TimJentzsch) [inline design/src/glossary.md:41]:
```suggestion
The degree to which game elements (usually [theme](glossary.md#theme) or [art](glossary.md#art-style)) feel like they belong together as part of a greater whole.
```

File: design/src/glossary.md [hunk 2]
```diff
@@ -11,34 +11,212 @@ Key concepts and terms from game design.
...
+
+### Constraint
+
+A limitation of the [design space](glossary.md#design-space), which must (or should) be respected when doign game design.
```
Comment (TimJentzsch) [inline design/src/glossary.md:45]:
```suggestion
A limitation of the [design space](glossary.md#design-space), which must (or should) be respected when doing game design.
```

File: design/src/glossary.md [hunk 3]
```diff
@@ -11,34 +11,212 @@ Key concepts and terms from game design.
...
+Depth is generally viewed as a positive attribute, while complexity is generally viewed as a negative attribute.
+Games which have high depth relative to their complexity are often considered **elegant.**
+
+Dependending on their player base, games have a varying **complexity budget.**
+This fuzzy term represents each player's ability to keep track of rules, learn new mechanics and reason effectively about their [choices](glossary.md#interesting-choices).
+If this budget is exceeded, players will be frustrated and confused by your game.
+
+### Diagetic
+
+A [mechanic] that is presented as occuring in the literal [game world](glossary.md#game-world).
```
Comment (TimJentzsch) [inline design/src/glossary.md:93]: A think the link to "mechanic" is missing here

**Review by TimJentzsch — COMMENTED** [review]
I did a first pass on typos.

I also think this needs to be rebased :)

File: design/src/glossary.md [hunk 4]
```diff
@@ -11,54 +11,268 @@ Key concepts and terms from game design.
...
 ### Upgrade
 
+An improvement to a [mechanic](glossary.md#mechanic).
+Upgrades are usually but not always selected by players to give them a sense of [agency](glossary.md#agency).
+
+### Worldbuilding
+
+The lore, places, people, [aesthetics](glossary.md#art-style) and internally consistent rules of your fictional universe.
+
 ### World generation
```
Comment (bzm3r) [inline design/src/glossary.md:276]:
```suggestion
### World generation

Usually refers to the [procedural generation](glossary.md#procedural-generation) of a game world, or parts of a game world, in opposition to the crafting of a world (or level) by hand.

Closely related to concepts of:
* [randomness](glossary.md#randomness): a pristine world provides a particular set of options for a player to begin engaging with, and these options should allow the player to meaningfully engage given the [mechanics](glossary.md#mechanic) available to them
* [plausibility/realism](glossary.md#plausibility-and-realism): a procedurally generated world must make sense given the game's [world building](glossary.md#worldbuilding)
```

File: design/src/glossary.md [hunk 5]
```diff
@@ -67,6 +281,18 @@ Terms defined here are standard elements of the factory builder genre.
...
+
+### Assembler
+
+A building or machine that can be used to process items via a [recipe](glossary.md#recipe).
+
+### Balancer
+
+Joins two or more item streams and evenly distributes items across them.
+
+Commonly combined with [balancers](glossary.md#balancer) or built out off [splitters](glossary.md#splitter).
```
Comment (bzm3r) [inline design/src/glossary.md:294]:
```suggestion
Commonly combined with [balancers](glossary.md#balancer) or built out of [splitters](glossary.md#splitter).
```

File: design/src/glossary.md [hunk 6]
```diff
@@ -255,23 +541,55 @@ In some games there are also many other entities to guide the behavior of large
...
 
+Dirt, but viewed scientifically.
+Useful for growing stuff.
+
+Soil has four main components, which vary in [water storage capacity](glossary.md#water-storage-capacity) and [drainage](glossary.md#water-drainage).
+
+1. Sand: large particles. Drains well, but water storage capacity suffers.
+2. Silt: medium particles. Pretty balanced.
+3. Clay: small particles. Drains terrible, but with great water storage capacity.
+4. Organic matter: used to be living stuff, carefully prepared and mixed in the soil. Great drainage and water storage, but decomposes.
```
Comment (bzm3r) [inline design/src/glossary.md:567]: Is it possible for something to be great at draining, but also be great at water storage?

Also, stuff like peat (or general bog-like stuff) is quite "clay-like" in behaviour but very organic?

I am not sure if organic matter should be a separate category, rather each one of sand/silt/clay can be made up, to varying degrees, of organic matter?

**Review by bzm3r — COMMENTED** [review]
Nice!

File: design/src/glossary.md [hunk 7]
```diff
@@ -11,54 +11,268 @@ Key concepts and terms from game design.
...
+
+While constraints are limiting, they're a powerful tool to create a more focused game and are great sparks for creativity.
+
+Constraints exist on a continuum between **soft** (flexible, can be ignored with a good enough reason) and **hard** (must be respected at all times).
+
+Constraints can be in [tension](glossary.md#tension) with each other.
+
+### Cycle (mechanics design)
+
+A set of elements of the same type that share strong mechanical (and often thematic) similarities.
```
[the remark this answers is not in the document]
Comment (alice-i-cecile) [reply design/src/glossary.md:65]: This is a term from Magic the Gathering design!

File: design/src/glossary.md [hunk 8]
```diff
@@ -11,54 +11,268 @@ Key concepts and terms from game design.
...
+
 ### Flavor text
 
-### Fun
+Writing (or sometimes other elements by analogy) that has no [mechanical](glossary.md#mechanic) impact or explanation,
+and instead serves to enhance the theme and tone of the game.
+
+### Game feel
+
+How satisfying and natural a game feels to play.
```
[the remark this answers is not in the document]
Comment (alice-i-cecile) [reply design/src/glossary.md:125]:
```suggestion
How satisfying, intuitive and natural a game feels to play.
Games with great game feel start to feel like extensions of the player's body as they master the game.
```

File: design/src/glossary.md [hunk 9]
```diff
@@ -11,54 +11,268 @@ Key concepts and terms from game design.
...
+In order for choices to be interesting, players must be able to:
+
+1. Understand the choices they have
+2. Understand the implication of each choice
+3. Care about the impact that the choice has on the game
+4. Decide differently based on game conditions
+
+Choices are said to be **balanced** if they provide roughly equivalent benefits (or drawbacks) to the players.
+
+### Lens
```
[the remark this answers is not in the document]
Comment (alice-i-cecile) [reply design/src/glossary.md:153]:
```suggestion
### Lens (game analysis)
```

File: design/src/glossary.md [hunk 10]
```diff
@@ -11,54 +11,268 @@ Key concepts and terms from game design.
...
+
+A perspective on the design of a game.
+See [the deck of lenses](https://schellgames.com/art-of-game-design) for an excellent set of questions to ask about your games.
+
 ### Loop
 
+A path through a [game system](glossary.md#system-game-design) that leads back to itself.
```
[the remark this answers is not in the document]
Comment (alice-i-cecile) [reply design/src/glossary.md:160]: Yeah, that's a much weaker definition IMO. Loops aren't "piles of mechanics that the players do a bunch": they need to actually close!

File: design/src/glossary.md [hunk 11]
```diff
@@ -11,54 +11,268 @@ Key concepts and terms from game design.
...
+
+Each step in a loop can be **reinforcing** (positive feedback) or **dampening** (negative feedback), resulting in an overall reinforcing or dampening cycle.
+
 ### Ludotography
 
+The telling of a story through gameplay.
```
[the remark this answers is not in the document]
Comment (alice-i-cecile) [reply design/src/glossary.md:166]:
```suggestion
The telling of a story through gameplay.
From [this video](https://www.youtube.com/watch?v=5ebIizhEgxg&t=521s).
```

File: design/src/glossary.md [hunk 12]
```diff
@@ -11,54 +11,268 @@ Key concepts and terms from game design.
...
+A rule about how the game works.
+
+### Narrative
+
+The story of a game.
+Unlike most traditional works of art, narratives are often implicit or branching in games.
+
 ### Plausibility and realism
 
+Fictional worlds are plausible if they have a set of intuitive, internally consistent rules.
```
[the remark this answers is not in the document]
Comment (alice-i-cecile) [reply design/src/glossary.md:180]:
```suggestion
Fictional worlds are plausible if they have a set of learnable, internally consistent rules.
```

File: design/src/glossary.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (bzm3r) [reply design/src/glossary.md]: @alice-i-cecile Can this become learnable too?

[the remark this answers is not in the document]
Comment (alice-i-cecile) [reply design/src/glossary.md]: No, that's definitely not the sense that people mean.

File: design/src/glossary.md [hunk 13]
```diff
@@ -11,62 +11,296 @@ Key concepts and terms from game design.
...
+
 ### Flavor text
 
-### Fun
+Writing (or sometimes other elements by analogy) that has no [mechanical](glossary.md#mechanic) impact or explanation,
+and instead serves to enhance the theme and tone of the game.
+
+### Game feel
+
+How satisfying, intuitive and natural a game feels to play.
```
Comment (alice-i-cecile) [inline design/src/glossary.md:125]:
```suggestion
How satisfying, smooth, intuitive and natural a game feels to play.
```

**Pull request merged**